### PR TITLE
Fix control merging when overriding child controls

### PR DESCRIPTION
**Pull request by jquick**
Signed-off-by: Jared Quick <jquick@chef.io>

Currently when you override a control in a wrapper only specific fields actually get overridden. This change adds in the functionality so refs, tags, and controls are merged/overridden accordingly. 

Fixes https://github.com/inspec/inspec/issues/3145
Fixes https://github.com/inspec/inspec/issues/2199

### CODE REVIEW

**Review by clintoncwolfe — APPROVED** [review]
Looks good - thanks @jquick !

File: test/functional/inspec_exec_test.rb [hunk 1]
```diff
@@ -329,6 +329,27 @@
...
+    let(:controls) { json['profiles'][0]['controls'] }
+    let(:override) { controls.select { |c| c['title'] == 'Profile 1 - Control 2-updated' }.first }
+
+    it 'completes the run with failed controls but no exception' do
+      out.stderr.must_be_empty
+      out.exit_status.must_equal 0
+      controls.count.must_equal 2
+
+      # check for json override
+      assert = "  control 'pro1-con2' do\n    impact 0.999\n    title 'Profile 1 - Control 2-updated'\n    desc 'Profile 1 - Control 2 description-updated'\n    tag 'password-updated'\n    ref 'Section 3.5.2.1', url: 'https://example.com'\n    describe file('/etc/passwd') do\n      it { should exist }\n    end\n  end\n"
```
Comment (jerryaldrichiii) [inline test/functional/inspec_exec_test.rb:344]: You could remove the leading white space...and likely the white space between the `\n`'s but I'll leave this up to you.

File: test/unit/mock/profiles/wrapper-override/controls/defaut.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+include_controls 'myprofile1' do
+  # skip this useless control
+  skip_control 'pro1-con3'
+
+  # fix this control from the profile we inherit
+  control 'pro1-con2' do
+    impact 0.999
```
Comment (jerryaldrichiii) [inline test/unit/mock/profiles/wrapper-override/controls/defaut.rb:7]: Is `0.999` valid? Thought it was to one decimal place.

**Review by jerryaldrichiii — APPROVED** [review]
Looks good to me, only some non-blocking questions.

Crazy amount of change just by setting one instance variable.

File: test/unit/mock/profiles/wrapper-override/controls/defaut.rb (hunk 2 above)
Comment (jquick) [reply test/unit/mock/profiles/wrapper-override/controls/defaut.rb:7]: impact can be whatever you want for now.

**Comment by alexpop** [comment]
thank you gents! 👏

**Pull request merged**